include/mata/nfa.hh

```diff
@@ -263,29 +263,296 @@ Mata::Parser::ParsedSection serialize(
     inline bool operator<=(const Move& rhs) const { return symbol <= rhs.symbol; }
-    inline bool operator>(const Move& rhs) const { return symbol > rhs.symbol; }
-    inline bool operator>=(const Move& rhs) const { return symbol >= rhs.symbol; }
     inline bool operator==(const Move& rhs) const { return symbol == rhs.symbol; }
     inline bool operator!=(const Move& rhs) const { return symbol != rhs.symbol; }
+    inline bool operator>(const Move& rhs) const { return symbol > rhs.symbol; }
+    inline bool operator>=(const Move& rhs) const { return symbol >= rhs.symbol; }
+
+    StateSet::iterator begin() { return targets.begin(); }
+    StateSet::iterator end() { return targets.end(); }
+
+    StateSet::const_iterator cbegin() const  { return targets.cbegin(); }
+    StateSet::const_iterator cend() const { return targets.cend(); }
 };
 
-/// List of transitions from a certain state. Each element holds transitions with a certain symbol.
-using Moves = Mata::Util::OrdVector<Move>;
-/// Transition relation for an NFA. Each index 'i' to the vector represents a state 'i' in the automaton.
-using TransitionRelation = std::vector<Moves>;
+/**
+ * Post is a data structure representing possible transitions over different symbols.
+ * It is an ordered vector containing possible Moves (i.e., pair of symbol and target states.
+ * Vector is ordered by symbols which are numbers.
+ */
+struct Post : private Util::OrdVector<Move> {
+    using iterator = Util::OrdVector<Move>::iterator;
+    using const_iterator = Util::OrdVector<Move>::const_iterator;
+
+    iterator begin() override { return Util::OrdVector<Move>::begin(); }
+    const_iterator begin() const override { return Util::OrdVector<Move>::begin(); }
+    iterator end() override { return Util::OrdVector<Move>::end(); }
+    const_iterator end() const override { return Util::OrdVector<Move>::end(); }
+
+    const_iterator cbegin() const override { return Util::OrdVector<Move>::cbegin(); }
+    const_iterator cend() const override { return Util::OrdVector<Move>::cend(); }
+
+    Post() = default;
+
+    virtual ~Post() = default;
+
+    const_iterator find(const Move& m) const override { return Util::OrdVector<Move>::find(m);}
+    iterator find(const Move& m) override { return Util::OrdVector<Move>::find(m);}
+
+    void insert(const Move& m) override { Util::OrdVector<Move>::insert(m); }
+
+    Move& operator[](Symbol s)
+    {
+        const iterator item = this->find(Move(s));
+        if (item == this->end())
+            throw std::runtime_error("Move not found");
+
+        return *item;
+    }
+
+    const Move& operator[](Symbol s) const
+    {
+        const auto item = this->find(Move(s));
+        if (item == this->end())
+            throw std::runtime_error("Move not found");
+
+        return *item;
+    }
+
+    const Move& back() const override { return Util::OrdVector<Move>::back(); }
+
+    void push_back(const Move& m) override{ Util::OrdVector<Move>::push_back(m); }
```

Comment:
should be insert? Call push_back if possible, else insert. Or just call OrdVector.insert,
 after OrdVector.insert it is modified to try to push_back if possible.